daml-foundations/daml-ghc/ghc-compiler/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -349,19 +349,32 @@ convertGenericTemplate env x
         let tplPrecondition = ETrue
         let tplAgreement = mkEmptyText
         let tplKey = Nothing
+        archive <- convertExpr env (Var archive)
         let convertGenericChoice :: [Var] -> ConvertM (TemplateChoice, [LF.Expr])
-            convertGenericChoice [controllers, action, exercise] = do
+            convertGenericChoice [consumption, controllers, action, exercise] = do
                 TContractId _ :-> _ :-> argType@(TConApp argTCon _) :-> TUpdate resType <- convertType env (varType action)
                 let chcLocation = Nothing
                 let chcName = ChoiceName $ T.intercalate "." $ unTypeConName $ qualObject argTCon
-                let chcConsuming = True
+                consumptionType <- case varType consumption of
+                    TypeCon (Is "NonConsuming") _ -> pure NonConsuming
```

Comment:
Adding more and more cases where we don’t at least match on the module name makes me sad. We should really fix this soon or at least do it properly for new code.

Reply:
I agree in general but I think this place is not particularly scary since the parser will generate these instances and hence it's under our control. At some point I will just kill the `Is` pattern synonym and force every pattern match to be proper.